sklearn/metrics/_scorer.py

```diff
@@ -368,28 +343,21 @@ def _score(self, method_caller, clf, X, y, sample_weight=None):
             raise ValueError("{0} format is not supported".format(y_type))
 
         if is_regressor(clf):
-            y_pred = method_caller(clf, "predict", X)
+            y_pred, _ = method_caller(clf, "predict", X)
         else:
+            pos_label = self._get_pos_label()
             try:
-                y_pred = method_caller(clf, "decision_function", X)
+                y_pred, _ = method_caller(
+                    clf, "decision_function", X, pos_label=pos_label
+                )
```

Comment:
Here, we could do ``y_pred, pos_label = method_caller(...)``, no ?
In which case it could make sense for ``_cached_call`` to return a tuple